common/sagas/schedule/paramsValidity.ts

```diff
@@ -0,0 +1,154 @@
+import { SagaIterator, delay } from 'redux-saga';
+import { select, fork, call, take, apply, put } from 'redux-saga/effects';
+import { getOffline, getNodeLib } from 'selectors/config';
+import {
+  ICurrentSchedulingToggle,
+  ICurrentWindowSize,
+  ICurrentWindowStart,
+  ICurrentScheduleType,
+  ICurrentScheduleTimezone,
+  ICurrentScheduleTimestamp,
+  ICurrentTimeBounty,
+  getSchedulingToggle,
+  getScheduleTimestamp,
+  getScheduleTimezone,
+  getScheduleType,
+  getWindowStart,
+  getWindowSize,
+  getTimeBounty,
+  getScheduleGasPrice,
+  getScheduleGasLimit,
+  getScheduleDeposit
+} from 'selectors/schedule/fields';
+import {
+  TypeKeys,
+  SetScheduleParamsValidityAction,
+  setScheduleParamsValidity
+} from 'actions/schedule';
+import { TypeKeys as TransactionTypeKeys } from 'actions/transaction';
+import { getData } from 'selectors/transaction/fields';
+import { getWalletInst } from 'selectors/wallet';
+import {
+  EAC_SCHEDULING_CONFIG,
+  calcEACEndowment,
+  getValidateRequestParamsData,
+  EAC_ADDRESSES,
+  parseSchedulingParametersValidity
+} from 'libs/scheduling';
+import { gasPriceToBase } from 'libs/units';
+import BN from 'bn.js';
+import { bufferToHex } from 'ethereumjs-util';
+import RequestFactory from 'libs/scheduling/contracts/RequestFactory';
+import { windowSizeBlockToMin, calculateWindowStart } from 'selectors/schedule/helpers';
+import { getCurrentTo, getCurrentValue } from 'selectors/transaction/current';
+
+export function* shouldValidateParams(): SagaIterator {
+  while (true) {
+    yield take([
+      TransactionTypeKeys.TO_FIELD_SET,
+      TransactionTypeKeys.DATA_FIELD_SET,
+      TypeKeys.CURRENT_TIME_BOUNTY_SET,
+      TypeKeys.WINDOW_SIZE_FIELD_SET,
+      TypeKeys.WINDOW_START_FIELD_SET,
+      TypeKeys.SCHEDULE_TIMESTAMP_FIELD_SET,
+      TypeKeys.TIME_BOUNTY_FIELD_SET,
+      TypeKeys.SCHEDULE_TYPE_SET,
+      TypeKeys.SCHEDULING_TOGGLE_SET,
+      TypeKeys.SCHEDULE_TIMEZONE_SET
+    ]);
+
+    yield call(delay, 250);
+
+    const isOffline: boolean = yield select(getOffline);
+    const schedulingToggle: ICurrentSchedulingToggle = yield select(getSchedulingToggle);
+    const scheduling = Boolean(schedulingToggle && schedulingToggle.value);
+
+    if (isOffline || !scheduling) {
```

Comment:
This can be computed in a selector instead

Reply:
We can see other places where it's used like this, I don't see really reason to extract this to another function.

example sagas with usage like this:
- node.ts
- gas.ts
- nonce.ts
- validationHelpers.ts